### PR TITLE
DFR-3223 Case submitted missing required information

### PR DESCRIPTION
**Pull request by so99y**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DFR-3223

### Checklist

- [X] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by al-hmcts — DISMISSED** [review]
LGTM!

**Review by al-hmcts — DISMISSED** [review]
Re-approve

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/utils/NullChecker.java [hunk 1]
```diff
@@ -12,6 +15,12 @@ private NullChecker() {
 
     @SuppressWarnings({"java:S3011", "java:S3864"})
     public static boolean anyNonNull(Object target) {
+        if (target == null) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/utils/NullChecker.java:18]: Although this code change looks innocuous it's not required and you're changing the function to have a dual purpose. The purpose of the method is to check if any of the fields in the target object are non-null. Now you're also making it check to see if target is a String and has some text. Why? You can use StringUtils.hasText directly in the client code.

Comment (so99y) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/utils/NullChecker.java]: @al-hmcts @jthmcts 
Updated my change accordingly - https://github.com/hmcts/finrem-case-orchestration-service/pull/1800/commits/10b9a2222d8b2bca7697b4e8d04b2800759108a2

**Review by jthmcts — DISMISSED** [review]

**Comment by so99y** [comment]
Fixed an issue found by Tom - [464a664](https://github.com/hmcts/finrem-case-orchestration-service/pull/1800/commits/464a6649466bbce6c406c5b636c08ec7fff1f6fa)
@jthmcts can I have a review on this please?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidator.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorEmail())) {
+                ret.add("Applicant solicitor's email is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorPhone())) {
+                ret.add("Applicant solicitor's phone is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorFirm())) {
+                ret.add("Applicant solicitor's name of your firm is required.");
+            }
+            if (!StringUtils.hasText(contactDetailsWrapper.getSolicitorName())) {
```
Comment (hamzahtahirhmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidator.java:41]: We can potentially make a method out of this. Something like this?

```
    private void validateField(String fieldValue, String fieldName, List<String> errors) {
        if (!StringUtils.hasText(fieldValue)) {
            errors.add(String.format("Applicant solicitor's %s is required.", fieldName));
        }
    }
```

Comment (so99y) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidator.java]: @hamzahtahirhmcts Good idea. Fixed by https://github.com/hmcts/finrem-case-orchestration-service/pull/1800#:~:text=the%20common%20logic-,909aac8,-Merge%20state

**Review by hamzahtahirhmcts — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidatorTest.java:28]: Not required

Comment (so99y) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/mandatorydatavalidation/ApplicantSolicitorDetailsValidatorTest.java]: Thanks

**Review by jthmcts — APPROVED** [review]

**Review by al-hmcts — DISMISSED** [review]

**Review by jthmcts — APPROVED** [review]

**Review by al-hmcts — APPROVED** [review]

**Pull request merged**